lms/djangoapps/course_home_api/progress/v1/views.py

```diff
@@ -122,11 +129,32 @@ def get(self, request, *args, **kwargs):
 
         enrollment_mode, _ = CourseEnrollment.enrollment_mode_for_user(request.user, course_key)
 
-        course_grade = CourseGradeFactory().read(request.user, course)\
+        # The block structure is used for both the course_grade and has_scheduled content fields
+        # So it is called upfront and reused for optimization purposes
+        collected_block_structure = get_block_structure_manager(course_key).get_collected()
+        course_grade = CourseGradeFactory().read(request.user, collected_block_structure=collected_block_structure)
+
+        # Get has_scheduled_content data
+        transformers = BlockStructureTransformers()
+        transformers += [start_date.StartDateTransformer()]
+        usage_key = collected_block_structure.root_block_usage_key
+        course_blocks = get_course_blocks(
+            request.user,
+            usage_key,
+            transformers=transformers,
+            collected_block_structure=collected_block_structure,
+            include_has_scheduled_content=True
+        )
+        has_scheduled_content = course_blocks.get_xblock_field(usage_key, 'has_scheduled_content')
+
+        # Get user_has_passing_grade data
+        user_has_passing_grade = False
+        if not request.user.is_anonymous:
```

Comment:
Is this check necessary? What's the value of `course_grade` if the request.user is anonymous?

Reply:
i think it makes sense to consistent with the existing definition of user_has_passing_grade
this won't set it if the user is anonymous though?